junit-common/src/main/java/org/jboss/weld/junit/MockBeanWithPriority.java

```diff
@@ -0,0 +1,147 @@
+package org.jboss.weld.junit;
+
+import javax.enterprise.context.Dependent;
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.inject.Any;
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.Prioritized;
+import javax.enterprise.inject.spi.Unmanaged;
+import javax.inject.Named;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Set;
+
+public class MockBeanWithPriority<T> extends AbstractMockBean<T> implements Prioritized {
```

Comment:
Why do we need this type? Couldn't we just enhance the existing `MockBean`?

Reply:
Dismissing this as I reworked it completely.